### DISCUSSION TITLE
combine 2 displays as one

### DISCUSSION THREAD

**Discussion by edward3h** [rm-hull/luma.core#245] [Q&A]
I'm building a device with 2 OLED displays side by side. I'd like to be able to use them as if they are one bigger display. Has anyone done something like this already?

If not I'm imagining I can make a device class that splits its display image in half and sends each to the underlying devices. Haven't dug into the details yet though.

**Comment by rm-hull** [comment]
A few folks have used 2 displays together for sure, but not exactly as you have described. I think it would be very easy to implement actually. 

You are exactly right: create a virtual device that extends ‘device’ class which takes two devices in its constructor, then override the ‘display’ method - this would use pillow to split the incoming image into two, one sent to the left devices display method, the other image sent to the right devices display method

**Reply by edward3h** [reply]
Finally got around to trying this, and I was able to get it working without too much effort! https://github.com/edward3h/luma_combine/blob/main/luma_combine/__init__.py

I copied and hacked the "welcome" example from luma.examples and have the text scrolling across both my screens. This highlights that the displays are not updating at the same time. I might dig into that later, but for my actual project the display is going to be more static so it's not going to show so much.
